### PR TITLE
Fix #41, #189 and #190 : updated agg methods docstring + solved ddof arg 

### LINKED ISSUE #41: Change signature of functions sum, mean, ...
Replace *args and **kwargs by equivalent arguments of Numpy functions.

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

**Comment by alixdamman** [comment]
Once this PR is merged (or just before the next release), we should build documentation with tag='latest' on readthedocs and see if everything is OK.

**Comment by gdementen** [comment]
The change in ddof needs a changelog entry. It also closes issue #190. I imagine (doc)tests will fail though because the numbers will not be the same.

**Comment by gdementen** [comment]
Also on what ground did you choose ddof=1 over ddof=0? Just because of Pandas or for another reason?

File: larray/core.py [hunk 1]
```diff
@@ -6625,29 +6626,29 @@ def percent(self, *axes):
...
+        def decorated(func):
+            def wrapper(self, *args, **kwargs):
+                keepaxes = kwargs.pop('keepaxes', False)
+                skipna = kwargs.pop('skipna', None)
+                if skipna is None:
+                    skipna = nanfunc is not None
+                if skipna and nanfunc is None:
+                    raise ValueError("skipna is not available for {}".format(func.__name__))
+                _npfunc = nanfunc if skipna else npfunc
+                extra_kwargs = {k: kwargs.pop(k) for k in list(kwargs.keys()) if k in _kwarg_agg.keys()}
```
Comment (gdementen) [inline larray/core.py]: where is _kwarg_agg defined?

File: larray/core.py [hunk 2]
```diff
@@ -6625,29 +6626,29 @@ def percent(self, *axes):
...
+                if 'percentile' in func.__name__:
+                    extra_kwargs['q'] = args[0]
```
Comment (gdementen) [inline larray/core.py]: This seems too specific. Wouldn't an some kind of extra argument (eg. fill_extra_kwargs) to _decorate_agg_method be more generic here? which would be given as a dict: {argname: argposition}. For example for percentile, that would be:
{'q': 0}?

File: larray/core.py [hunk 3]
```diff
@@ -6713,9 +6714,11 @@ def method(self, *args, **kwargs):
         >>> # or equivalently
         >>> # barr.all('a0,a1>>a01;a2,a3>>a23')
         """.format(_doc_agg_method("AND reduction", kwargs="out,skipna,keepaxes"))
+        pass
```
Comment (gdementen) [inline larray/core.py]: This is very strange to use a decorator over a function and not use the decorated function at all, except its name and signature. Is there no other way?

**Comment by alixdamman** [comment]
It seems that using S_{n-1} is a common practice in statistics. 
https://en.wikipedia.org/wiki/Standard_deviation#Corrected_sample_standard_deviation

I am not sure why Numpy set ddof to 0 be default.

File: larray/core.py (hunk 1 above)
Comment (alixdamman) [inline larray/core.py]: just before finction _doc_agg_method which is defined above class LArray

File: larray/core.py (hunk 3 above)
Comment (alixdamman) [inline larray/core.py]: Yes, there is another way. Need some additional work.

File: larray/core.py (hunk 1 above)
Comment (gdementen) [inline larray/core.py]: It did not appear in the diff, so I thought it was some broken code. Sorry about this!

File: larray/core.py (hunk 2 above)
Comment (alixdamman) [inline larray/core.py]: I discover that explicit arguments like ddof=1 are not passed to wrapper. I will try to add extra_args=() and extra_kwargs={} as arguments to _decorate_agg_method.

**Comment by alixdamman** [comment]
Need review before going further. 
Using decorator was a bad idea. It is a nightmare to handle optional arguments. 

I propose something that is somewhat close to what Pandas does --> see end of file 
https://github.com/pandas-dev/pandas/blob/v0.19.2/pandas/core/generic.py

It is also a nightmare to find simple doctests for var, std and describe with ddof=1 by default...

**Comment by alixdamman** [comment]
Do not review... I'm working on it.

**Comment by alixdamman** [comment]
Overriding signature in doctstring is working with Sphinx --> 
http://larray-test.readthedocs.io/en/fix_agg_methods_docstring_and_ddof_arg_41_189/api.html#la-agg
but not sure that PyCharm is smart enough to show 'new' signature with Ctrl+Q and Ctrl+P

**Comment by gdementen** [comment]
The result is great. You just need to include both the "extra arguments fix" and default ddof change in the changelog.

**Comment by gdementen** [comment]
LGTM

**Pull request merged**